src/compiler/factory.ts

```diff
@@ -1260,11 +1268,11 @@ namespace ts {
                     foundUseStrict = true;
                 }
 
-                target.push(source[i]);
+                target.unshift(source[i]);

```

Comment:
I see what you mean now. But there are the same and it will not matter right? also if the prologue is sprinkle around the body function, it will be aggregate to the top so that should be ok as well? 

The function is not necessary call at the beginning of an empty array. Like [below ](https://github.com/Microsoft/TypeScript/pull/7953/files#diff-2320501f9a3b426f8ee7654c548418e9R1221) the target statements-array is not empty